nb2kg/nb2kg/__init__.py

```diff
@@ -16,12 +16,25 @@ def _jupyter_server_extension_paths():
 def load_jupyter_server_extension(nb_app):
     '''Loads server extension.'''
     nb_app.log.info('Loaded nb2kg extension')
+
     # TODO: There is no clean way to override existing handlers that are already
     # registered with the Tornado application.  The first handler to match the
     # URL will handle the request, so we must prepend our handlers to override 
     # the existing ones.
     web_app = nb_app.web_app
-    pattern, handlers = web_app.handlers[0]
+
+    # Detect tornado compatibility.  The 'handlers' list was "moved" (and
+    # transformed) in tornado >= 4.5.  Check for its existence and continue the
+    # load accordingly.  In 4.5, a "rules" list is used and now resides under
+    # the wildcard_router.  Fortunately, URLSpec derives from Rule (as of 4.5)
+    # so much of the previous code can be retained.
+
+    # Detect older version of tornado
+    if getattr(web_app, 'handlers', None):  # tornado < 4.5
```

Comment:
Probably better written as:

`if hasattr(web_app, 'handlers'): # tornado < 4.5

since the actually attribute value is unused.

Reply:
Nice catch on `hasattr` vs. `getattr` - update pushed.